### DISCUSSION TITLE
Passing a Task object to a function

### DISCUSSION THREAD

**Discussion by IRodriguezSC** [sadieblue/agilityblue.workspace-scripts#27] [Q&A] [answered]
Does anyone know how to pass a task object as a parameter.  For example:

$task   = Get-Task $taskId -IncludeForms

function ([Parameter(Mandatory=$true)][parameter_type]$taskParam) {
 
#---your code that uses the task (ie myID = $taskParam.Id) #
}

I can pass the task ID and use Get-Task but wondering if Ican actually pass a task object already created.

Thanks,
-IR

**Comment by mrabb24** [comment, marked as answer]
Hello,

PowerShell lets you define a function with statically assigned values:

```powershell
function My-Function($taskParam) {
    Write-Output "taskParam id is $($taskParam.TaskId)"
}

$taskId = 1

# Get a task
$task = Get-Task -Id $taskId -IncludeForms

# Call your function
My-Function -taskParam $task
```

However, they don't really recommend static assignment. Instead, you can use parameters:

```powershell
function My-Function {
    param (
        $taskParam
    )

    Write-Output "taskParam id is $($taskParam.TaskId)"
}

$taskId = 1

# Get a task
$task = Get-Task -Id $taskId -IncludeForms

# Call your function
My-Function -taskParam $task
```

Either way will work, and the parameters don't need type information, unlike a strongly typed language does (like Java or C#).

[More information on the subject](https://learn.microsoft.com/en-us/powershell/scripting/learn/ps101/09-functions?view=powershell-7.4#parameters)

Matt

**Reply by IRodriguezSC** [reply]
Great thanks, will give this a try.

**Reply by IRodriguezSC** [reply]
Just wanted to let you know that I have everything rolling now.  Thanks!

**Comment by IRodriguezSC** [comment]
This has been resolved by applying mrabb24 suggestions.
